include/nbl/video/IDeviceMemoryBacked.h

```diff
@@ -13,12 +13,38 @@ namespace nbl::video
 struct NBL_API2 ICleanup
 {
     virtual ~ICleanup() = 0;
+
+    std::unique_ptr<ICleanup> next;
+
+    static void chain(std::unique_ptr<ICleanup>& first, std::unique_ptr<ICleanup>&& next)
+    {
+        if (first)
+            return chain(first->next, std::move(next));
+        first = std::move(next);
+    }
 };
 
 //! Interface from which resources backed by IDeviceMemoryAllocation inherit from
 class IDeviceMemoryBacked : public virtual core::IReferenceCounted
 {
     public:
+        //! Flags for imported/exported allocation
+        enum E_EXTERNAL_HANDLE_TYPE : uint32_t
+        {
+            EHT_NONE = 0,
+            EHT_OPAQUE_FD = 0x00000001,
+            EHT_OPAQUE_WIN32 = 0x00000002,
+            EHT_OPAQUE_WIN32_KMT = 0x00000004,
+            EHT_D3D11_TEXTURE = 0x00000008,
+            EHT_D3D11_TEXTURE_KMT = 0x00000010,
+            EHT_D3D12_HEAP = 0x00000020,
+            EHT_D3D12_RESOURCE = 0x00000040,
+            EHT_DMA_BUF = 0x00000200,
+            EHT_HOST_ALLOCATION = 0x00000080,
+            EHT_HOST_MAPPED_FOREIGN_MEMORY = 0x00000100,
+            EHT_RDMA_ADDRES = 0x00001000,
```

Comment:
only leave uncommented the ones you'll implement/add a codepath for

Reply:
btw IDeviceMemoryBacked is not inherited from by semaphores or device memory allocations, are you sure all these enums are applicable to buffers and images?

Reply:
Semaphores have seperate flags. Buffers, images and memory allocations use the same bitflags.